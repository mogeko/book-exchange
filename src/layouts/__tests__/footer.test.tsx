import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Footer from "@/layouts/footer";

describe("Footer", () => {
  it("create a footer without author", () => {
    render(<Footer time={1997} />);
    expect(document.querySelector(".footer-center p")!.textContent).toBe(
      "Copyright © 1997 - 2022, All right reserved."
    );
  });

  it("create a footer with author", () => {
    render(<Footer time={2022} author="Zheng Junyi" />);
    expect(document.querySelector(".footer-center p")!.textContent).toBe(
      "Copyright © 2022, All right reserved by Zheng Junyi."
    );
  });

  it("snapshot a footer", () => {
    const { container } = render(<Footer time={1997} author="Zheng Junyi" />);
    expect(container).toMatchSnapshot();
  });
});
