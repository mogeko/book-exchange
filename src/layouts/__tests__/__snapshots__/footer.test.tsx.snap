// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer create a footer with author 1`] = `
<div>
  <footer
    aria-label="Footer"
    class="footer footer-center p-4 bg-base-300 text-base-content"
  >
    <div>
      <p
        aria-label="copyright"
        role="presentation"
      >
        Copyright © 
        2022
        , All right reserved
         by Zheng Junyi
        .
      </p>
    </div>
  </footer>
</div>
`;

exports[`Footer create a footer without author 1`] = `
<div>
  <footer
    aria-label="Footer"
    class="footer footer-center p-4 bg-base-300 text-base-content"
  >
    <div>
      <p
        aria-label="copyright"
        role="presentation"
      >
        Copyright © 
        1997 - 2022
        , All right reserved
        .
      </p>
    </div>
  </footer>
</div>
`;
